### PR TITLE
Aptos telemetry web service

### PR DESCRIPTION
**Pull request by ibalajiarun**
### Description

This PR introduces the Aptos Telemetry web service. It can be used to expose various endpoints to collect telemetry information from Aptos validators and nodes. Currently, the service can authenticate validators and issue them with JWT tokens that they can inject into the Authorization header to post authorized telemetry.

For more details, refer to the [design proposal](https://www.notion.so/aptoslabs/Telemetry-Ingestion-Service-Proposal-da06afd80f744c72a919420878c36de1).

### Test Plan

I have two tests that tries the authentication flow.

`cargo test`

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2460)

### CODE REVIEW

File: crates/aptos-telemetry-service/src/tests/mod.rs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+mod test_context;
+mod auth_test;
+
+/// Returns the name of the current function. This macro is used to derive the name for the golden
+/// file of each test case.
+#[macro_export]
+macro_rules! current_function_name {
```
Comment (geekflyer) [inline crates/aptos-telemetry-service/src/tests/mod.rs:7]: let's not copy and paste this but rather put this into some central utility crate/library and import it from there.

File: crates/aptos-telemetry-service/src/auth.rs [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+}
+
+pub async fn handle_auth(
+    context: Context,
+    body: AuthRequest,
+) -> anyhow::Result<impl Reply, Rejection> {
+    let client_init_message = &body.handshake_msg;
+
+    // verify that this is indeed our public key
+    if body.server_public_key != context.noise_config().public_key() {
```
Comment (alinush) [inline crates/aptos-telemetry-service/src/auth.rs:35]: What is the server checking here exactly? That the client (validator) included the server's correct Noise public key in the HTTP request?

**Review by alinush — CHANGES_REQUESTED** [review]
Please address `auth.rs:35`

File: crates/aptos-telemetry-service/src/auth.rs (hunk 2 above)
Comment (ibalajiarun) [reply crates/aptos-telemetry-service/src/auth.rs:39]: Yes, we are trying to make sure that the client is using the correct public key for the server and inform the client if its using a wrong key.

Comment (ibalajiarun) [reply crates/aptos-telemetry-service/src/auth.rs:39]: Updated the code comment to be clear.

**Review by alinush — APPROVED** [review]

**Review by rustielin — APPROVED** [review]

**Pull request merged**